### DISCUSSION TITLE
Easier way to filter workspaces in `npm` commands

### DISCUSSION THREAD

**Discussion by trusktr** [npm/rfcs#735] [RRFC]
I would like to be able to do this:

```sh
npm install --workspace 'packages/!(some-package)'
```

which would install for all workspaces except `some-package`. Basically it would be nice if the pattern syntax had more power. If it does, I haven't been able to guess what it is, and I haven't seen it in the docs.

**Comment by kylerneva** [comment]
npm already provides workspace filtering through --workspace and --workspaces flags, but it doesn’t currently support advanced patterns like exclusions. Adding more complex pattern syntax would increase command parsing complexity, and the current design favors explicit workspace targeting over implicit pattern matching.
